apps/studio/components/interfaces/Storage/EditBucketModal.tsx

```diff
@@ -106,42 +100,105 @@ export const EditBucketModal = ({ visible, bucket, onClose }: EditBucketModalPro
     if (bucket === undefined) return console.error('Bucket is required')
     if (ref === undefined) return console.error('Project ref is required')
 
-    updateBucket(
-      {
+    form.clearErrors()
+    setIsUpdating(true)
+
+    // Client-side validation: Check if bucket limit exceeds global limit
+    // [Joshen] Should shift this into superRefine in the form schema
+    if (
+      values.has_file_size_limit &&
+      values.formatted_size_limit !== undefined &&
+      data?.fileSizeLimit
+    ) {
+      const bucketLimitInBytes = convertToBytes(
+        values.formatted_size_limit,
+        selectedUnit as StorageSizeUnits
+      )
+
+      if (bucketLimitInBytes > data.fileSizeLimit) {
+        form.setError('formatted_size_limit', { type: 'manual', message: 'exceed_global_limit' })
+        return setIsUpdating(false)
+      }
+    }
+
+    try {
+      const result = await updateBucket({
```

Comment:
ahh JFYI we should be using the react query mutations, not the update methods directly
so that the related data after updating the bucket gets invalidated and refreshed

although good point on this - we shouldn't export these methods unless needed

Reply:
hmm the docs there are probably useful to understand how the react queries work
but the way we use react query isnt exactly like how the docs describe it haha

probably easier to reference our existing react query hooks, or our templates [here](https://github.com/supabase/supabase/blob/master/apps/studio/data/__templates/README.md#L16) - although we haven't relooked at the templates README in a long while so just feel free to call out if any parts of it might be confusing or seems wrong 😅

Reply:
Looks great from a glance. Will do, though. Thanks!